tests/unit/plugins/modules/test_lambda_layer.py

```diff
@@ -0,0 +1,497 @@
+                        ]
+                    }
+                ]
+            }
+        ),
+        (
+            {
+                "name": "testlayer",
+                "version": -1
+            },
+            [
+                {
+                    'compatible_runtimes': ["python3.7"],
+                    'created_date': "2022-09-29T10:31:35.977+0000",
+                    'layer_version_arn': "arn:aws:lambda:eu-west-2:123456789012:layer:testlayer:2",
+                    "license_info": "MIT",
+                    'version': 2,
+                    'compatible_architectures': [
+                        'arm64'
+                    ]
+                },
+                {
+                    "created_date": "2022-09-29T10:31:26.341+0000",
+                    "description": "lambda layer first version",
+                    "layer_version_arn": "arn:aws:lambda:eu-west-2:123456789012:layer:testlayer:1",
+                    "version": 1
+                }
+            ],
+            [
+                call(LayerName='testlayer', VersionNumber=2),
+                call(LayerName='testlayer', VersionNumber=1)
+            ],
+            {
+                "changed": True,
+                "layer_versions": [
+                    {
+                        'compatible_runtimes': ["python3.7"],
+                        'created_date': "2022-09-29T10:31:35.977+0000",
+                        'layer_version_arn': "arn:aws:lambda:eu-west-2:123456789012:layer:testlayer:2",
+                        "license_info": "MIT",
+                        'version': 2,
+                        'compatible_architectures': [
+                            'arm64'
+                        ]
+                    },
+                    {
+                        "created_date": "2022-09-29T10:31:26.341+0000",
+                        "description": "lambda layer first version",
+                        "layer_version_arn": "arn:aws:lambda:eu-west-2:123456789012:layer:testlayer:1",
+                        "version": 1
+                    }
+                ]
+            }
+        )
+    ]
+)
+def test_delete_layer(params, api_result, calls, ansible_result):
+
+    lambda_client = MagicMock()
+    lambda_client.delete_layer_version.return_value = None
+    lambda_layer.list_layer_versions = MagicMock()
+
+    lambda_layer.list_layer_versions.return_value = api_result
+    result = lambda_layer.delete_layer_version(lambda_client, params)
+    assert result == ansible_result
+
+    lambda_layer.list_layer_versions.assert_called_once_with(
+        lambda_client, params.get("name")
+    )
+
+    if not calls:
+        lambda_client.delete_layer_version.assert_not_called()
+    else:
+        lambda_client.delete_layer_version.assert_has_calls(calls, any_order=True)
+
+
+def test_delete_layer_check_mode():
+
+    lambda_client = MagicMock()
+    lambda_client.delete_layer_version.return_value = None
+    lambda_layer.list_layer_versions = MagicMock()
```

Comment:
If you do this, you modify the `list_layer_versions` of the original copy of the module. It's better to avoid this because it can can impact other tests later and it's hard to troubleshoot the problem.

You can use either `patch` or `monkeypatch` to address this. e.g:
https://gist.github.com/goneri/28ed9d22c1419bb8f052f5350e4b88be#file-test_lambda_layer-py-L251

They will ensure the attribute is mocked only for this test.

Reply:
updated, thanks